x/fees/keeper/fees.go

```diff
@@ -0,0 +1,189 @@
+package keeper
+
+import (
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/tharsis/evmos/v3/x/fees/types"
+)
+
+// GetAllFees - get all registered DevFeeInfo instances
+func (k Keeper) GetAllFees(ctx sdk.Context) []types.DevFeeInfo {
+	feeInfos := []types.DevFeeInfo{}
+
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, types.KeyPrefixFee)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		contractAddress := common.BytesToAddress(iterator.Key())
+		deployerAddress := sdk.AccAddress(iterator.Value())
+		withdrawalAddress, _ := k.GetWithdrawal(ctx, contractAddress)
+		feeInfo := types.DevFeeInfo{
+			ContractAddress: contractAddress.String(),
+			DeployerAddress: deployerAddress.String(),
+			WithdrawAddress: withdrawalAddress.String(),
+		}
```

Comment:
I think the empty address string still prints the bech32 HRP, so we need to change this to: 

```suggestion
		withdrawalAddress, hasWithdrawAddr := k.GetWithdrawal(ctx, contractAddress)
		feeInfo := types.DevFeeInfo{
			ContractAddress: contractAddress.String(),
			DeployerAddress: deployerAddress.String(),
		}
		
		if hasWithdrawAddr {
			feeInfo.WithdrawAddress = withdrawalAddress.String()
		}
```

Reply:
If a contract is registered, the `GetWithdrawal` function always returns an address. If there is no withdrawal address, it returns the deployer.
I preferred to have this logic in the lowest level keeper getter, so we don't need to worry about empty addresses in the rest of the code.
`GetWithdrawal` returns `(sdk.AccAddress, bool)` - the boolean is false only in the case where the contract is not registered.

Reply:
Fixed in https://github.com/tharsis/evmos/pull/436/commits/404b219ca3740a5fb46595e330fc9817ceff2584